public/components/incontext_insight/generate_popover_body.tsx

```diff
@@ -55,10 +56,19 @@ export const GeneratePopoverBody: React.FC<{
     const getMonitorType = async () => {
       const context = await incontextInsight.contextProvider?.();
```

Comment:
Not related to this PR, could you store this context in a variable so we don't need to invoke this provider multiple times?(It got 3 invoking for now.) contextProvider for alerting is a heavy operation, it will execute a dsl and a ppl each time.

Reply:
Sure, let me see how to redesign this and do it in a separate PR in order not to have any influence on upcoming release.